add examples for strapi middlewares/controllers to graphql docs (#1680)
strapi/documentation: docusaurus/docs/dev-docs/plugins/graphql.md

##### Policies

> [Policies](/dev-docs/backend-customization/policies) can be applied to a GraphQL resolver through the `resolversConfig.[MyResolverName].policies` key. Only difference between the GraphQL and REST implementation is that the `config` key becomes `options`.

Reviewer: Where did you encounter `options`? :thinking: 
GraphQL => https://github.com/strapi/strapi/blob/main/packages/plugins/graphql/server/services/content-api/policy.js#L18
REST => https://github.com/strapi/strapi/blob/main/packages/core/strapi/lib/services/server/policy.js#L16

Author: @Convly It is for middleware
https://github.com/strapi/strapi/blob/6c81e74ebc7b39f541e864a92907bc530a48c842/packages/plugins/graphql/server/services/content-api/wrap-resolvers.js#L38-L40C38
Made a mistake that I thought it where policies.

> [Policies](/dev-docs/backend-customization/policies) can be applied to a GraphQL resolver through the `resolversConfig.[MyResolverName].policies` key. Only difference between the GraphQL and REST implementation is that the `config` key becomes `options`.

Reviewer 2: ```suggestion
[Policies](/dev-docs/backend-customization/policies) can be applied to a GraphQL resolver through the `resolversConfig.[MyResolverName].policies` key.
```

               */ 
              return context.parent !== undefined;
            }
            /**

>              * Uses a already created policy in strapi.

Reviewer 2: ```suggestion
             * Uses an already created policy in strapi.
```

             */
            "api::model.policy-name",

            /**

>              * Uses a already created policy in strapi with custom config.

Reviewer 2: ```suggestion
             * Uses an already created policy in strapi with custom config.
```

             */

>             {name:"api::model.policy-name" options: {/* all config values I want to pass to the strapi middleware */} },

Reviewer: ```suggestion
            {name:"api::model.policy-name", config: {/* all config values I want to pass to the strapi policy */} },
```

               */ 
              return context.parent !== undefined;
            }
            /**

>              * Uses a already created policy in strapi.

Reviewer 2: ```suggestion
             * Uses an already created policy in strapi.
```

             */
            "api::model.policy-name",

            /**

>              * Uses a already created policy in strapi with custom config.

Reviewer 2: ```suggestion
             * Uses an already created policy in strapi with custom config.
```

             */

>             {name:"api::model.policy-name" options: {/* all the configuration values to pass to the strapi middleware */} },

Reviewer: ```suggestion
            {name:"api::model.policy-name", config: {/* all the configuration values to pass to the strapi policy */} },
```

</details>

##### Middlewares

> [Middlewares](/dev-docs/backend-customization/middlewares) can be applied to a GraphQL resolver through the `resolversConfig.[MyResolverName].middlewares` key. The only difference between the GraphQL and REST implementations is that the `config` key becomes `options`.  

Reviewer: Yeah this one is on me, it should have been config since the beginning but now we have options :man_facepalming:

              return resolve(parent, ...rest);
            }
            /**
             * Basic middleware example #4

>              * Uses a already created middleware in strapi.

Reviewer 2: ```suggestion
             * Uses an already created middleware in strapi.
```

             */
            "api::model.middleware-name",

            /**
             * Basic middleware example #5

>              * Uses a already created middleware in strapi with custom config.

Reviewer 2: ```suggestion
             * Uses an already created middleware in strapi with a custom configuration
```

             */

>             { name: "api::model.middleware-name" options: { /* all config values I want to pass to the strapi middleware */ } },

Reviewer: ```suggestion
            { name: "api::model.middleware-name", options: { /* all config values I want to pass to the strapi middleware */ } },
```

              return resolve(parent, ...rest);
            }

            /**
             * Basic middleware example #4
             * Uses a already created middleware in strapi.
             */
            "api::model.middleware-name",

            /**
             * Basic middleware example #5

>              * Uses a already created middleware in strapi with custom config.

Reviewer 2: ```suggestion
             * Uses a middleware already created in Strapi with a custom configuration
```

             */

>             {name:"api::model.middleware-name" options: {/* all the configuration values to pass to the middleware */} },

Reviewer: ```suggestion
            {name:"api::model.middleware-name", options: {/* all the configuration values to pass to the middleware */} },
```
